### PR TITLE
PLATUI-1261: Add a RichErrorSummary

### PR DESCRIPTION
**Pull request by matthewmascord**

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcScriptsSpec.scala [hunk 1]
```diff
@@ -38,11 +38,12 @@ class hmrcScriptsSpec
   "hmrcScripts" should {
     "include the hmrc-frontend script tag" in {
       val hmrcScripts = app.injector.instanceOf[HmrcScripts]
+      hmrcfrontend.RoutesPrefix.setPrefix("/foo-service/hmrc-frontend")
       val scripts     = hmrcScripts().select("script")
 
       scripts                   should have size 1
       scripts.first.attr("src") should fullyMatch regex
-        """/assets/hmrc-frontend-\d+.\d+.\d+.min.js""".r
+        """/foo-service/hmrc-frontend/assets/hmrc-frontend-\d+.\d+.\d+.min.js""".r
```
Comment (oscarduignan) [inline src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcScriptsSpec.scala:46]: @matthewmascord just to check, is this change intentional or part of something else?

Comment (matthewmascord) [reply src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcScriptsSpec.scala:46]: There is a race condition around this because the reverse router is statically initialised and it was failing locally. So this forces it to have a certain value.

**Review by JoPintoPaul — DISMISSED** [review]

File: src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Implicits.scala [hunk 2]
```diff
@@ -223,6 +223,58 @@ trait Implicits {
...
+    *
+    * For example:
+    * {{{
+    *   @govukErrorSummary(ErrorSummary().withFormTextErrors(dateInputForm, mapping = Map("date" -> "date.day")))
+    * }}}
+    */
+  implicit class RichErrorSummary[T](errorSummary: ErrorSummary)(implicit val messages: Messages) {
+    import uk.gov.hmrc.govukfrontend.views.Implicits.RichFormErrors
+
+    def withFormHtmlErrors(form: Form[T]): ErrorSummary =
```
[the remark this answers is not in the document]
Comment (matthewmascord) [reply src/main/scala/uk/gov/hmrc/hmrcfrontend/views/Implicits.scala:242]: What about `withFormErrorsAsText` and `withFormErrorsAsHtml`?

**Review by oscarduignan — APPROVED** [review]
